dress_up/dress_up.lua

```diff
@@ -0,0 +1,269 @@
+local player = require('player')
+local target = require('target')
+local packets = require('packets')
+local entities = require('entities')
+local settings = require('settings')
+
+local ui = require('core.ui')
+local command = require('core.command')
+
+local defaults = {
+    auto_update = true,
+
+    dress = {
+    --[[
+        <player_name> = {
+            race_model_id = 7 -- id of model to use.
+        }
+    ]]--
+    },
+    
+    ui = {
+        x = 10,
+        y = 10,
+    },
+
+    blink = {
+        follow = {
+            always = false,
+            target = false
+        },
+        others = {
+            always = false,
+            target = false
+        },
+        party = {
+            always = false,
+            target = false
+        },
+        self = {
+            always = false,
+            target = false
+        }
+    }
+}
+
+local options = settings.load(defaults)
+
+local settings_dialog = {}
+local settings_dialog_state = {
+    title = 'Dress Up Settings',
+    style = 'normal',
+    x = options.ui.x,
+    y = options.ui.y,
+    width = 179,
+    height = 96,
+    resizable = false,
+    moveable = true,
+    closable = true,
+}
+
+-- Command Arg Types
+-- Create and Register types
+local arg_lookups = {
+    conditions = {
+        ['target'] = true,
+        ['always'] = true,
+        ['combat'] = true,
+        ['all'] = true,
+    },
+    groups = {
+        ['others'] = 'others',
+        ['o'] = 'others',
+        ['follow'] = 'follow',
+        ['f'] = 'follow',
+        ['party'] = 'party',
+        ['p'] = 'party',
+        ['self'] = 'self',
+        ['s'] = 'self',
+        ['all'] = 'all',
+        ['a'] = 'all',
+    },
+
+    bool = {
+        ['1'] = true,
+        ['true'] = true,
+        ['t'] = true,
+        ['yes'] = true,
+        ['y'] = true,
+        ['on'] = true,
+        ['0'] = false,
+        ['false'] = false,
+        ['f'] = false,
+        ['no'] = false,
+        ['n'] = false,
+        ['off'] = false,
+    },
+}
+
+command.arg.register('name', '<name:string(%a+)>')
+command.arg.register('condition', '<condition:one_of(all,always,target)>')
+command.arg.register('model_name', '<model_name:one_of(face,race,head,body,hands,legs,feet,main,sub,ranged)>')
+
+
+command.arg.register_type('lookup_boolean', {
+    check = function(str)
+        local bool = arg_lookups.bool[str]
+        if bool ~= nil then
+            return bool
+        end
+
+        error('Expected one of \'' .. table.concat(arg_lookups.bool, ',') .. '\', received \'' .. str .. '\'.')
+    end
+})
+command.arg.register_type('lookup_group', {
+    check = function(str)
+        local group = arg_lookups.groups[str]
+        if group ~= nil then
+            return group
+        end
+
+        error('Expected one of \'' .. table.concat(arg_lookups.groups, ',') .. '\', received \'' .. str .. '\'.')
+    end
+})
+
+-- Addon Command Handlers
+-- Five commands
+local du = command.new('du')
```

Comment:
Here it says `/du` now, so I assume the `/do` up there was a typo?

Reply:
correct